src/tekton/taskruntemplate.ts

```diff
@@ -0,0 +1,93 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+
+import * as yaml from 'js-yaml';
+import { newK8sCommand, TektonNode } from '../tkn';
+import * as vscode from 'vscode';
+import { Task, TaskRunTemplate } from '../tekton';
+import { cli } from '../cli';
+
+async function getTask(name: string): Promise<Task> {
+  const task = await cli.execute(newK8sCommand(`get task ${name} -o json`));
+  const taskData = JSON.parse(task.stdout);
+  return taskData;
+}
+
+function getInputsOutputs(task: Task, taskRunTemplate: TaskRunTemplate): void {
+  const resources = {};
+  if (task.spec.resources?.outputs && task.spec.resources?.outputs.length !== 0) {
+    const outputs = [];
+    task.spec.resources.outputs.map((value) => {
```

Comment:
Please read [map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) doc
You use `map` in wrong(anti-pattern) way